integration/tests/cook/test_basic.py

```diff
@@ -3006,6 +3006,26 @@ def test_kubernetes_disallowed_var_names(self):
         finally:
             util.kill_jobs(self.cook_url, [job_uuid1, job_uuid2])
 
+    @unittest.skipUnless(util.using_kubernetes(), 'Test requires kubernetes')
+    def test_kubernetes_checkpointing(self):
+        docker_image = util.docker_image()
+        container = {'type': 'docker',
+                     'docker': {'image': docker_image}}
+        try:
+            command_disabled = 'bash -c \'if [[ "${COOK_CHECKPOINT_MODE:-none}" == "none" ]] && [[ "${COOK_CHECKPOINT_PERIOD_SEC:-zzz}" == "zzz" ]]; then exit 0; else exit 1; fi\''
+            job_uuid_disabled, resp_disabled = util.submit_job(self.cook_url, command=command_disabled, container=container)
+            self.assertEqual(201, resp_disabled.status_code)
+            command_enabled = 'bash -c \'if [[ "${COOK_CHECKPOINT_MODE}" == "auto" ]] && [[ "${COOK_CHECKPOINT_PERIOD_SEC}" == "555" ]] && [[ "${COOK_CHECKPOINT_PRESERVE_PATH_0}" == "p1" ]] && [[ "${COOK_CHECKPOINT_PRESERVE_PATH_1}" == "p2" ]]; then exit 0; else exit 1; fi\''
+            job_uuid_enabled, resp_enabled = util.submit_job(self.cook_url, command=command_enabled, container=container,
+                                                             checkpoint={"mode": "auto",
+                                                                         "periodic-options": {"period-sec": 555},
+                                                                         "options": {"preserve-paths": ["p2", "p1"]}})
+            self.assertEqual(201, resp_enabled.status_code)
+            util.wait_for_instance(self.cook_url, job_uuid_disabled, status='success')
+            util.wait_for_instance(self.cook_url, job_uuid_enabled, status='success')
+        finally:
+            util.kill_jobs(self.cook_url, [job_uuid_disabled, job_uuid_enabled])
```

Comment:
Should we add a test for the `periodic` and `preemptible` modes?

Reply:
don't think so - just testing that the different types of elements of the configuration are working